c10/util/TypeIndex.h

```diff
@@ -31,36 +43,91 @@ namespace detail {
 #error "You're running a too old version of GCC. We need GCC 5 or later."
 #endif
 
+inline constexpr string_view extract(
+    string_view prefix,
+    string_view suffix,
+    string_view str) {
+#if !defined(__CUDA_ARCH__) // CUDA doesn't like std::logic_error in device code
```

Comment:
Umm, and why should this ever be run on device?

Reply:
it shouldn't be but CUDA seems to want to compile it for device code just in case.

Reply:
This is what `__host__` is for